scripts/registries-test.py

```diff
@@ -0,0 +1,422 @@
+    result = subprocess.run(
+        ["op", "item", "list", "--vault", vault_name, "--format", "json"],
+        capture_output=True,
+        text=True,
+        check=True,
+    )
+
+    items = json.loads(result.stdout)
+    updated_env = env.copy()
+
+    for item in items:
+        item_id = item["id"]
+        item_title = item["title"]
+
+        # Only process items that match the registry naming pattern
+        if item_title.startswith("UV_TEST_"):
+            # Extract the registry name (e.g., "AWS" from "UV_TEST_AWS")
+            registry_name = item_title[8:]  # Remove "UV_TEST_" prefix
+
+            # Get the item details
+            item_details = subprocess.run(
+                ["op", "item", "get", item_id, "--format", "json"],
+                capture_output=True,
+                text=True,
+                check=True,
+            )
+
+            item_data = json.loads(item_details.stdout)
+
+            username = None
+            password = None
+            url = None
+            pkg = None
+
+            if "fields" in item_data:
+                for field in item_data["fields"]:
+                    if field.get("id") == "username":
+                        username = field.get("value")
+                    elif field.get("id") == "password":
+                        password = field.get("value")
+                    elif field.get("label") == "url":
+                        url = field.get("value")
+                    elif field.get("label") == "pkg":
+                        pkg = field.get("value")
+            if username:
+                updated_env[f"UV_TEST_{registry_name}_USERNAME"] = username
+            if password:
+                updated_env[f"UV_TEST_{registry_name}_TOKEN"] = password
+            if url:
+                updated_env[f"UV_TEST_{registry_name}_URL"] = url
+            if pkg:
+                updated_env[f"UV_TEST_{registry_name}_PKG"] = pkg
+
+            print(f"Added 1Password credentials for {registry_name}")
+
+    return updated_env
+
+
+def get_registries(env: Dict[str, str]) -> Dict[str, str]:
+    pattern = re.compile(r"^UV_TEST_(.+)_URL$")
+    registries: Dict[str, str] = {}
+
+    for env_var, value in env.items():
+        match = pattern.match(env_var)
+        if match:
+            registry_name = match.group(1).lower()
+            registries[registry_name] = value
+
+    return registries
+
+
+def setup_test_project(
+    registry_name: str, registry_url: str, project_dir: str, requires_python: str
+):
+    """Create a temporary project directory with a pyproject.toml"""
+    pyproject_content = f"""[project]
+name = "{registry_name}-test"
+version = "0.1.0"
+description = "Test registry"
+requires-python = ">={requires_python}"
+
+[[tool.uv.index]]
+name = "{registry_name}"
+url = "{registry_url}"
+default = true
+"""
+    pyproject_file = Path(project_dir) / "pyproject.toml"
+    pyproject_file.write_text(pyproject_content, encoding="utf-8")
```

Comment:
We don't need the `encoding="utf-8"` here. https://peps.python.org/pep-0686/ will solve this fully, but we're not writing non-ascii here.